rust/lance-io/src/object_writer.rs

```diff
@@ -50,15 +69,38 @@ impl ObjectWriter {
                     location: location!(),
                 })?;
 
+        let writer = Arc::new(Mutex::new(Pin::new(writer)));
+
+        // If background task encounters an error, we use a channel to send the error
+        // to the main task.
+        let (error_sender, background_error) = tokio::sync::oneshot::channel();
+
+        let writer_ref = writer.clone();
+        let background_flusher = tokio::task::spawn(async move {
+            loop {
+                tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+                match writer_ref.lock().unwrap().flush().now_or_never() {
```

Comment:
What happens if we call flush after a writer is finished writing / closed?

Also, what is the normal exit path for this task?  It looks like it can only exit this loop if `flush` returns an error.

Reply:
Right now calling flush after it's done seems to be a no-op, but there's no strong guarantee of that in the API. We should abort this task in the shutdown method, I think.